arbnode/seq_coordinator.go

```diff
@@ -671,46 +675,68 @@ func (c *SeqCoordinator) Start(ctxIn context.Context) {
 	}
 }
 
-func (c *SeqCoordinator) waitForHandoff(ctx context.Context) string {
-	var nextChosen string
+// Calls check() on retryInterval until it returns a non-default value, which is then returned.
+// On context timeout, this returns the default value.
+func waitFor[T comparable](ctx context.Context, retryInterval time.Duration, check func() T) T {
+	var empty T
 	for {
-		var err error
-		nextChosen, err = c.CurrentChosenSequencer(ctx)
-		if err == nil && nextChosen != "" && nextChosen != c.config.MyUrl() {
-			return nextChosen
+		result := check()
+		if result != empty {
+			return result
 		}
 		select {
 		case <-ctx.Done():
-			return ""
-		case <-time.After(c.config.RetryInterval):
+			return empty
+		case <-time.After(retryInterval):
 		}
 	}
 }
 
 func (c *SeqCoordinator) PrepareForShutdown() {
+	// normal context will be closed, use parent context
+	parentCtx := c.StopWaiter.GetParentContext()
+	handoffCtx, cancel := context.WithTimeout(c.StopWaiter.GetContext(), c.config.ShutdownHandoffTimeout)
```

Comment:
handoffCtx can be based on GetContext() instead of GetParentCtx() because StopAndWait wasn't called yet but parent is also fine..

Reply:
handoffCtx is used later after we do call StopAndWait. We could use two different contexts but I figured one timeout for both is better.